nengo/builder/optimizer.py

```diff
@@ -84,7 +84,7 @@ def optimize(model, dg, max_passes=None):
         before = len(dg)
 
         with Timer() as t:
-            single_pass(only_merge_ops_with_view)
+            sig_replacements.update(single_pass(only_merge_ops_with_view))
```

Comment:
I though about returning this and I decided against it because this make the function call return a value *and* have side effects. I prefer to only do one or the other because
* We're having side effects already (modifying dg for example), why not be consistent and do everything as a side effect?
* Returning a value might give the impression that there are no side effects and I saw bugs (in other projects) introduced because of that.